lib/service/supervisor.go

```diff
@@ -252,7 +251,7 @@ func (s *LocalSupervisor) RemoveService(srv Service) error {
 	defer s.Unlock()
 	for i, el := range s.services {
 		if el == srv {
-			s.services = append(s.services[:i], s.services[i+1:]...)
+			s.services = slices.Delete(s.services, i, i+1)
```

Comment:
nit: this probably woul dbe better as a slices.DeleteFunc to replace the for loop.

Reply:
I'll pass on this one, since using `DeleteFunc` would change the semantics a bit (we wouldn't be able to do the early return, and we'd have to do more book-keeping to have the same logging emitted).